File: frontend/src/services/paymentService.js
import { paymentApi } from './api'

// Process a payment for an order
export const processPayment = (data) =>
  paymentApi.post('/api/payments', data)

// Get payment details by payment ID
export const getPaymentById = (id) =>
  paymentApi.get(`/api/payments/${id}`)

// Get payment for a specific order
export const getPaymentByOrderId = (orderId) =>
  paymentApi.get(`/api/payments/order/${orderId}`)

// Get all payments (admin)
export const getAllPayments = () =>
  paymentApi.get('/api/payments')

File: backend/payment-service/src/main/java/com/ecommerce/paymentservice/service/PaymentService.java
package com.ecommerce.paymentservice.service;

import com.ecommerce.paymentservice.dto.PaymentDTO;
import com.ecommerce.paymentservice.dto.ProcessPaymentRequest;

import java.util.List;

/**
 * PaymentService interface - defines all payment operations.
 */
public interface PaymentService {

    /** Process a payment for an order */
    PaymentDTO processPayment(ProcessPaymentRequest request);

    /** Get payment details by payment ID */
    PaymentDTO getPaymentById(Long id);

    /** Get payment for a specific order */
    PaymentDTO getPaymentByOrderId(Long orderId);

    /** Get all payments (admin) */
    List<PaymentDTO> getAllPayments();

    /** Update payment status (e.g. for refunds) */
    PaymentDTO updatePaymentStatus(Long id, String status);
}
